test/roby_app/test_plugin.rb

```diff
@@ -212,6 +212,143 @@ def perform_app_assertion(result)
+                    client.create_log_dir(
+                        @ps_log_dir, Roby.app.time_tag,
+                        { "parent" => Roby.app.app_metadata }
+                    )
+                end
+
+                def close_process_servers
+                    @process_servers.each do |name, thread, client|
+                        client.close
+                        Syskit.conf.remove_process_server(name)
+                        thread.raise Interrupt
+                        thread.join
+                    end
+                end
+
+                def create_test_file(ps_log_dir)
+                    logfile = File.join(ps_log_dir, "logfile.log")
+                    File.open(logfile, "wb") do |f|
+                        # create random 5 MB file
+                        f.write(SecureRandom.random_bytes(547))
+                    end
+                    logfile
+                end
+
+                def start_local_transfer_server(tmp_root_ca)
+                    @tmp_server_dir = make_tmpdir
+                    @user = "test.user"
+                    @password = "password123"
+                    @log_transfer_server = LogTransferIntegration::LocalLogTransferServer.new(
+                        @tmp_server_dir, @user, @password,
+                        tmp_root_ca.signed_certificate
+                    )
+                end
+
+                def wait_for_upload_completion(poll_period: 0.01, timeout: 1)
+                    deadline = Time.now + timeout
+                    loop do
+                        if Time.now > deadline
+                            flunk("timed out while waiting for upload completion")
+                        end
+
+                        state = @client.log_upload_state
+                        return state if state.pending_count == 0
+
+                        sleep poll_period
+                    end
+                end
+
+                def assert_upload_succeeds
+                    wait_for_upload_completion.each_result do |r|
+                        flunk("upload failed: #{r.message}") unless r.success?
+                    end
+                end
+
+                before do
+                    # Initializing Process Server
+                    Syskit.conf.only_load_models = false
+                    @process_servers = []
+                    create_process_server("test_ps")
+                end
+
+                after do
+                    close_process_servers
+                end
+
+                it "uploads file from Process Server" do
+                    @ps_logfile = create_test_file(@ps_log_dir)
+                    path = File.join(app.log_dir, "logfile.log")
+                    refute File.exist?(path)
+                    @client = Plugin.send_file_transfer_command("test_ps", @ps_logfile)
+                    assert_upload_succeeds
+                    assert_equal File.read(path), File.read(@ps_logfile)
+                end
```

Comment:
Would be better to put the actual test code first. The helpers are meant to be "implementation details"